packages/gateway-client/src/worker/p2p-client/stream-factory.ts

```diff
@@ -5,16 +5,110 @@ import type { P2pClient } from '.';
 import { ServerPeerStatus } from '../../worker-messaging';
 import { logger } from '../logging';
 import status from '../status';
+import { decode } from '../p2p-fetch/lob-enc';
+import { Deferred } from '../p2p-fetch/p2p-request';
```

Comment:
I'd actually prefer that we copy the class instead of re-using it, just because it is pretty simple and I don't want to introduce "util" type stuff (i.e. the way that we're also copying `waitFor` all over the place).